packages/valory/skills/mech_interact_abci/behaviours/response.py

```diff
@@ -200,22 +255,59 @@ def _get_response_hash(self) -> WaitableConditionType:
+
+        # Check if request IDs were successfully determined
+        if request_id_bytes is None or request_id_for_specs is None:
+            self.context.logger.error(
+                "Could not determine the request ID to use for fetching the response."
+            )
+            return False
+
         self.context.logger.info(
-            f"Filtering the mech's events from block {self.from_block} "
-            f"for a response to our request with id {request_id!r}."
-        )
-        result = yield from self.mech_contract_interact(
-            contract_callable="get_response",
-            data_key="data",
-            placeholder=get_name(MechResponseBehaviour.response_hex),
-            request_id=request_id,
-            from_block=self.from_block,
-            chain_id=self.params.mech_chain_id,
+            f"Filtering the Mech's Deliver events from block {self.from_block} "
+            f"for a response to request with id (bytes32) 0x{request_id_bytes.hex()} or (int) {request_id_for_specs}"
         )
 
+        # Conditionally call get_response based on whether the marketplace (and thus mech_mm) is used
+        if self.params.use_mech_marketplace:
+            # Use mech_mm ABI (MechMM.contract_id) and bytes32 request ID
+            self.context.logger.info(
+                f"Using Mech Marketplace flow: Calling get_response with bytes32 request ID 0x{request_id_bytes.hex()} using MechMM ABI."
+            )
+            result = yield from self.contract_interact(
+                performative=ContractApiMessage.Performative.GET_RAW_TRANSACTION,
+                contract_address=self.params.mech_contract_address,  # Target the mech_mm contract address
+                contract_public_id=MechMM.contract_id,  # Use MechMM ABI
+                contract_callable="get_response",
+                data_key="data",
+                placeholder=get_name(MechResponseBehaviour.response_hex),
+                request_id=request_id_bytes,  # Use bytes32 request ID
+                from_block=self.from_block,
+                chain_id=self.params.mech_chain_id,
+            )
+        else:
+            # Use legacy mech ABI (self.params.mech_contract_id) and int request ID
+            self.context.logger.info(
+                f"Using legacy Mech flow: Calling get_response with int request ID {request_id_for_specs} using legacy Mech ABI."
+            )
+            # Note: We rely on _mech_contract_interact using the correct legacy mech ABI via self.params.mech_contract_id
+            result = yield from self._mech_contract_interact(
+                contract_callable="get_response",
+                data_key="data",
+                placeholder=get_name(MechResponseBehaviour.response_hex),
+                request_id=request_id_for_specs,  # Use integer request ID for legacy mech
+                from_block=self.from_block,
+                chain_id=self.params.mech_chain_id,
+            )
```

Comment:
Please return the initialized generator from a method/property and then consume here to get the result.